src/Confluent.Kafka.Avro/AvroSerializer.cs

```diff
@@ -40,118 +36,25 @@ namespace Confluent.Kafka.Serialization
     {
-        private SpecificWriter<T> avroWriter;
-
-        private HashSet<string> topicsRegistered = new HashSet<string>();
+        private bool autoRegisterSchema = true;
+        private int initialBufferSize = DefaultInitialBufferSize;
+        private bool isKeySerializer;
+        private ISchemaRegistryClient schemaRegistryClient;
 
         private bool disposeClientOnDispose;
 
-        private string writerSchemaString { get; set; }
+        private IAvroSerializerImpl<T> serializerImpl;
 
         private const string InitialBufferSizePropertyName = "avro.serializer.buffer.bytes";
 
         private const string AutoRegisterSchemaPropertyName = "avro.serializer.auto.register.schemas";
 
-        /// <summary>
-        ///     The SchemaId corresponding to type <see cref="T"/>, or null if 
-        ///     the serializer has yet to be invoked.
-        /// </summary>
-        public int? SchemaId { get; private set; } = null;
-
-        /// <summary>
-        ///	    The <see cref="ISchemaRegistryClient"/> instance used for communication
-        ///	    with Confluent Schema Registry.
-        /// </summary>
-        public ISchemaRegistryClient SchemaRegistryClient { get; private set; }
-
         /// <summary>
         ///     The default initial size (in bytes) of buffers used for message 
         ///     serialization.
         /// </summary>
         public const int DefaultInitialBufferSize = 128;
 
-        /// <summary>
-        ///     True if the serializer will attempt to auto-register un-recognized schemas
-        ///     with Confluent Schema Registry, false if not.
-        /// </summary>
-        public bool AutoRegisterSchema { get; private set; } = true;
-
-        /// <summary>
-        ///     Initial size (in bytes) of the buffer used for message serialization.
-        /// </summary>
-        /// <remarks>
-        ///     Use a value high enough to avoid resizing of buffer, but small enough
-        ///     to avoid excessive memory use. Inspect the size of byte array returned by
-        ///     <see cref="Serialize(string, T)"/> to estimate an appropriate value.
-        ///     Note: each call to serialize creates a new buffer.
-        /// </remarks>
-        public int InitialBufferSize { get; private set; } = DefaultInitialBufferSize;
-
-        /// <summary>
-        ///	    True if this serializer is used for serializing Kafka message keys,
-        ///	    false if it is used for serializing Kafka message values.
-        /// </summary>
-        public bool IsKey { get; private set; }
-
-        /// <summary>
-        ///     The avro schema used to write values of type <see cref="T"/>
-        /// </summary>
-        public Avro.Schema WriterSchema { get; private set; }
-
-        private void Initialize()
-        {
-            Type writerType = typeof(T);
-            if (typeof(ISpecificRecord).IsAssignableFrom(writerType) || writerType.IsSubclassOf(typeof(SpecificFixed)))
```

Comment:
the logic is wrong here for `SpecificFixed` so I removed it. we can add this later.